src/Coordination/Changelog.cpp

```diff
@@ -1937,6 +1970,9 @@ try
         }
     }
 
+    if (remove_logs_before_index)
+        removeAllLogFilesBefore(remove_logs_before_index);
```

Comment:
Why do we need to remove them if have a code branch which handles such situation?